lib/utils/database.dart

```diff
@@ -1,36 +1,55 @@
-import '../models/user.dart';
-import '../models/contact.dart';
-import '../models/conversation.dart';
-import '../models/message.dart';
-import '../models/device.dart';
+import 'package:chirp_nets/models/user.dart';
+import 'package:chirp_nets/models/conversation.dart';
+import 'package:chirp_nets/models/contact.dart';
+import 'package:chirp_nets/models/message.dart';
+import 'package:chirp_nets/models/device.dart';
 import 'package:path/path.dart';
 import 'package:sqflite/sqflite.dart';
-import 'dart:async';
+
+final List<String> migrations = [
+  "CREATE TABLE users(id INTEGER PRIMARY KEY, name TEXT);",
+  "CREATE TABLE device(id INTEGER PRIMARY KEY, userId INTEGER, deviceRSSI TEXT, FOREIGN KEY(userId) REFERENCES users(id));",
+  "CREATE TABLE conversations(id INTEGER PRIMARY KEY, userId INTEGER, name TEXT, FOREIGN KEY(userId) REFERENCES users(id));",
+  "CREATE TABLE messages(id INTEGER PRIMARY KEY, conversationId INTEGER, message TEXT, createdAt TEXT, FOREIGN KEY(conversationId) REFERENCES conversations(id));",
+  "CREATE TABLE contact(id INTEGER PRIMARY KEY, conversationId INTEGER, name TEXT, deviceId TEXT, FOREIGN KEY(conversationId) REFERENCES conversations(id));",
+  "ALTER TABLE device ADD COLUMN name;",
+  "ALTER TABLE messages ADD COLUMN createdBy INTEGER; ALTER TABLE messages add FOREIGN KEY(createdBy) REFERENCES users(id)",
+  "ALTER TABLE contact ADD COLUMN userId INTEGER; ALTER TABLE contact add FOREIGN KEY(userId) REFERENCES users(id)",
+];
 
 Future<Database> getDatabase() async {
   final Future<Database> database = openDatabase(
     join(await getDatabasesPath(), 'chirpnets.db'),
     onCreate: (db, version) {
-      return db.execute(
-        """
-          CREATE TABLE users(id INTEGER PRIMARY KEY, name TEXT); 
-          CREATE TABLE device(id INTEGER PRIMARY KEY, userId INTEGER, deviceRSSI TEXT, FOREIGN KEY(userId) REFERENCES users(id));
-          CREATE TABLE conversations(id INTEGER PRIMARY KEY, userId INTEGER, name TEXT, FOREIGN KEY(userId) REFERENCES users(id));
-          CREATE TABLE messages(id INTEGER PRIMARY KEY, conversationId INTEGER, message TEXT, createdAt TEXT, FOREIGN KEY(conversationId) REFERENCES conversations(id));
-          CREATE TABLE contact(id INTEGER PRIMARY KEY, conversationId INTEGER, name TEXT, FOREIGN KEY(conversationId) REFERENCES conversations(id));
-        """,
-      );
+      for (var script in migrations) {
+        db.execute(script);
+      }
+    },
+    onUpgrade: (db, oldVersion, newVersion) async {
+      var batch = db.batch();
+      print(oldVersion);
+      print(newVersion);
+      for (var i = oldVersion - 1; i == newVersion - 1; i++) {
+        batch.execute(migrations[i]);
+      }
+      await batch.commit();
     },
-    version: 1,
+    version: migrations.length,
```

Comment:
This seems incorrect. Say we have 5 migrations, okay so version is 5, we get a new migration, okay version is 6. We then remove a migration, version goes back to 5, HOWEVER this is a new version of the code. Am I misunderstanding the version values usage?

Reply:
We shouldn't really be removing migrations, its supposed to be a sort log of the database developments that can be reproduced/reversed if needed. That being said, you're right that we shouldn't be using the number of migrations as the version number it should be done based on update to the db

Reply:
this wasnt reversed. was it supposed to be?